File: app/src/main/java/com/androidessence/cashcaretaker/fragments/DatePickerFragment.java
package com.androidessence.cashcaretaker.fragments;

import android.app.DatePickerDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;

import java.util.Calendar;
import java.util.Date;

/**
 * Dialog that allows a user to select a date.
 *
 * Created by adammcneilly on 10/15/15.
 */
public class DatePickerFragment extends DialogFragment{
    private static final String ARG_DATE = "dateArg";

    public static DatePickerFragment newInstance(Date date){
        DatePickerFragment fragment = new DatePickerFragment();
        Bundle args = new Bundle();
        args.putSerializable(ARG_DATE, date);
        fragment.setArguments(args);
        return fragment;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        Date mDate = (Date) getArguments().getSerializable(ARG_DATE);

        Calendar calendar = Calendar.getInstance();
        calendar.setTime(mDate);

        // Use mDate
        assert mDate != null;
        int year = calendar.get(Calendar.YEAR);
        int month = calendar.get(Calendar.MONTH);
        int day = calendar.get(Calendar.DAY_OF_MONTH);

        return new DatePickerDialog(getActivity(), (DatePickerDialog.OnDateSetListener)getTargetFragment(), year, month, day);
    }
}

File: app-refresh/src/main/java/com/adammcneilly/cashcaretaker/DatePickerFragment.kt
package com.adammcneilly.cashcaretaker

import android.app.DatePickerDialog
import android.app.Dialog
import android.os.Bundle
import android.support.v4.app.DialogFragment
import java.util.*


/**
 * Dialog fragment that allows user to pick a date.
 */
class DatePickerFragment : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val dateArg = arguments?.getSerializable(ARG_DATE) as Date

        val calendar = Calendar.getInstance()
        calendar.time = dateArg

        // Use mDate
        val year = calendar.get(Calendar.YEAR)
        val month = calendar.get(Calendar.MONTH)
        val day = calendar.get(Calendar.DAY_OF_MONTH)

        return DatePickerDialog(activity, targetFragment as DatePickerDialog.OnDateSetListener, year, month, day)
    }

    companion object {
        private val ARG_DATE = "dateArg"

        fun newInstance(date: Date): DatePickerFragment {
            val args = Bundle()
            args.putSerializable(ARG_DATE, date)

            val fragment = DatePickerFragment()
            fragment.arguments = args
            return fragment
        }
    }
}